common/psbt_open.c

```diff
@@ -0,0 +1,378 @@
+#include "common/psbt_open.h"
+#include <assert.h>
+#include <bitcoin/psbt.h>
+#include <bitcoin/script.h>
+#include <bitcoin/tx.h>
+#include <ccan/asort/asort.h>
+#include <ccan/ccan/mem/mem.h>
+#include <common/utils.h>
+
+bool psbt_get_serial_id(struct wally_unknowns_map *map, u16 *serial_id)
+{
+	u8 *key = psbt_make_key(tmpctx, PSBT_TYPE_SERIAL_ID, NULL);
+	size_t value_len;
+	void *result = psbt_get_unknown(map, key, &value_len);
+	if (!result)
+		return false;
+	assert(value_len == sizeof(*serial_id));
```

Comment:
I worry this is a vector by which bad PSBTs can crash us.  Perhaps return false in this case?